res/controllers/Denon-HC1000S-scripts.js

```diff
@@ -0,0 +1,869 @@
+            print("VinylControl Toggle is triggered");
+        } else {
+            engine.setValue("[VinylControl]", "Toggle", false);
+        }
+    } else {
+        if ((status & 0xF0) === 0x90) {
+            engine.setValue(group, "reverseroll", true);
+        } else {
+            engine.setValue(group, "reverseroll", false);
+        }
+    }
+};
+
+DenonHC1000S.LoadTrack = function(midino, control, value, status, group) {
+    if (!DenonHC1000S.shift) {
+        print("group is " + (group));
+        engine.setValue(group, "LoadSelectedTrack", 1);
+    } else {
+        print("shift & trackLoad => InstantDouble");
+        switch (group) {
+        case "[Channel1]":
+            engine.setValue(group, "CloneFromDeck", 2);
+            break;
+        case "[Channel2]":
+            engine.setValue(group, "CloneFromDeck", 1);
+            break;
+        case "[Channel3]":
+            engine.setValue(group, "CloneFromDeck", 4);
+            break;
+        case "[Channel4]":
+            engine.setValue(group, "CloneFromDeck", 3);
+            break;
+        }
+    }
+};
+
+DenonHC1000S.ledHotCueAndSampler = function(channel, control, value, status, group) {
+    var ledChannel = DenonHC1000S.getLedChannelByGroup(group); // 0xB0 or
+    if (DenonHC1000S.shiftSampleLeft || DenonHC1000S.shiftSampleRight) { //show Sampler/Play/Cue info instead of HotCue
+        var samplerLeds = [0x15, 0x17, 0x19, 0x65, 0x67, 0x69]; //,0x1B,0x40,0x20
+        print("ledChannel = " + ledChannel);
+        for (var index = 0, count = samplerLeds.length; index < count; index++) {
+            DenonHC1000S.handleLed(ledChannel, (engine.getValue("[Sampler" + (index + 1) + "]", "track_loaded") === 1), samplerLeds[index], 0x4C, 0x4B);
+            print("LED sampler is " + (index));
+        }
+        //Play Button on Shift-Cue-1
+        ledChannel = DenonHC1000S.getLedChannelByGroup(group);
+        //Channel 1+3
+        DenonHC1000S.handleLed((ledChannel), (engine.getValue("[Channel1]", "track_loaded") === 1), 0x11, 0x4C, 0x4B);
+        DenonHC1000S.handleLed((ledChannel), (engine.getValue("[Channel1]", "cue_indicator") === 1), 0x13, 0x4C, 0x4B);
+        //Channel 2+4
+        DenonHC1000S.handleLed((ledChannel), (engine.getValue("[Channel2]", "track_loaded") === 1), 0x61, 0x4C, 0x4B);
+        DenonHC1000S.handleLed((ledChannel), (engine.getValue("[Channel2]", "cue_indicator") === 1), 0x63, 0x4C, 0x4B);
+
+        // hotcues
+    } else {
+        ledChannel = DenonHC1000S.getLedChannelByGroup(group);
+        //Channel 1+3
+        var cueLeds = [0x11, 0x13, 0x15, 0x17, 0x19]; //,0x1B,0x40,0x20
+        print("LED group1 is " + (group));
+        for (index = 0, count = cueLeds.length; index < count; index++) {
+            if (group !== "[Channel3]") {
+                DenonHC1000S.handleLed(0xB0, (engine.getValue("[Channel1]", "hotcue_" + (index + 1) + "_position") !== -1), cueLeds[index], 0x4A, 0x4B);
```

Comment:
```suggestion
                DenonHC1000S.handleLed(0xB0, engine.getValue("[Channel1]", "hotcue_" + (index + 1) + "_enabled"), cueLeds[index], 0x4A, 0x4B);
```
You might prefer the `_enabled` CO here